pkgs/tools/system/hddfancontrol/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, python3Packages, fetchFromGitHub, hddtemp, hdparm, smartmontools }:
+
+python3Packages.buildPythonPackage rec {
+  pname = "hddfancontrol";
+  version = "1.5.1";
+
+  src = fetchFromGitHub {
+    owner = "desbma";
+    repo = pname;
+    rev = version;
+    sha256 = "0b2grf98qnikayn18xll01dkm5pjpcjxdffgx1nyw9s0gqig8dg0";
+  };
+
+  propagatedBuildInputs = [
+    python3Packages.python-daemon
+    hddtemp
+    hdparm
+    smartmontools
+  ];
+
+  postInstall = ''
```

Comment:
The upstream systemd service has `After=hddtemp.service`[1].  Where is that service from?  It is not on my system.

[1]: https://github.com/desbma/hddfancontrol/blob/ea4e8a113a6b70c8a9299cba31bb7712ec80ca3d/systemd/hddfancontrol.service#L3C7-L3C22

Reply:
There exists a nixos module for hddtemp: https://github.com/NixOS/nixpkgs/blob/nixos-23.05/nixos/modules/hardware/sensor/hddtemp.nix

I think it should be OK to have an After= relationship with a service that may not exist.  After= only has an effect if two units are started together, according to the [systemd.unit docs]( https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Before=), and if hddtemp is enabled together with hddfancontrol then it will be beneficial for hddtemp to be started first.